proto/poktroll/service/relay.proto

```diff
@@ -22,6 +23,10 @@ message RelayRequestMetadata {
   // application has delegated to. The signature is made using the ring of the
   // application in both cases.
   bytes signature = 2;
+
+  // The supplier address the relay is sent to. It is being used on the RelayMiner to 
+  // route to the correct supplier.
```

Comment:
This is a little weird.

I'm not opposed to it, but we're changing one of the most fundamental structures in the protocol (that'll be in every single leaf) solely for the reason of `It is being used on the RelayMiner to  route to the correct supplier`.

@red-0ne Do you have thoughts here?
@okdas can you add a `TODO_IN_THIS_PR: Re-evaluate` so we don't forget.

Reply:
If we do keep this, we'll also need to update the on-chain logic.

We'll need to verifythat the signer of the response equals to the public key of the address here.

If this is not done, we're going to find ourselves in a place where weird shit can happen between the metadata and the signatures.

Reply:
Two ways I can think of to solve that:
1. Remove the supplier address from the protobuf and rely on a different mechanism to select the supplier. I suggest using HTTP headers. Here's the example on the `shannon-sdk` side:
<img width="638" alt="Screenshot 2024-06-06 at 2 48 07 PM" src="https://github.com/pokt-network/poktroll/assets/4950477/ddffde64-5c59-4c81-b1ea-cf8275c8fe37">
And relayminer:
<img width="815" alt="Screenshot 2024-06-06 at 2 48 24 PM" src="https://github.com/pokt-network/poktroll/assets/4950477/e29d395d-ba54-4f0e-b6b9-cb303ea270fd">

2. Keep the supplier address in protobuf. We can introduce a proto that would be included in `RelayRequest` but would not be included in `SessionTree`. I am not sure exactly how yet, but there must be a way.

Also, a related question. That address that is being sent along with a relay is basically a recommendation to which supplier to route the request to. RelayMiner could potentially use a different supplier, that is also in the session, to serve the request instead. Correct?

Reply:
As discussed in [1], adding the supplier's address to the `RelayRequest.Meta` seemed to be the most reliable way to route requests.

> We'll need to verifythat the signer of the response equals to the public key of the address here.

Absolutely! The relay miner also needs to check that the `RelayRequest.SupplierAddress` matches a supplier that it manages.

@okdas, Adding a header filed in the request to identify the supplier seems like an implicit protocol requirement that lives outside of the protobuf. I prefer keep everything explicit and contained in the `RelayRequest` protobuf definition.

[1] https://github.com/pokt-network/poktroll/issues/528#issuecomment-2111043405

Reply:
@red-0ne after sleeping on this thought, I 100% agree. I added a check on a RelayMiner - if the supplier address is missing on the incoming request - it'll throw an error.

The other solution would be to adopt Morse ideology, where one virtual HTTP server/domain can serve requests only from one signing key (key1 -> supplier1.relayminers.com, key2 -> supplier2.relayminers.com). That wouldn't require any changes to the on-chain logic but would be an additional step in hydrating the configs on RelayMiner, and perhaps some checks. I personally don't see issues with that, and this design is community comfortable with.